lib/smppex/session.ex

```diff
@@ -130,6 +130,13 @@ defmodule SMPPEX.Session do
               | {:ok, [Pdu.t()], state}
               | {:stop, reason, state}
 
+  @doc """
+  Invoked when the session is about to be stopped due to a timeout.
+
+  The return value will be used as the exit reason of the session process. The default implementation returns `{:timers, reason}`.
+  """
+  @callback handle_timeout(timeout_reason :: term, state) :: exit_reason :: term
```

Comment:
I added exact type for timeout reasons in https://github.com/funbox/smppex/commit/21bc56e433b3c6fb0becddb5fbbe06edcedc40c2

Maybe we now specify `timeout_reason :: SMPPEX.SMPPTimers.stop_reason` so that it is easier to implement `handle_timeout` callback?